### DISCUSSION TITLE
Add IPAM system

### DISCUSSION THREAD

**Discussion by mdepedrof** [tteck/Proxmox#1327] [Ideas]
would be great if can add a netbox (a powerful network and ipam manager) to the scripts.
https://hub.docker.com/r/linuxserver/netbox#!

If this is complicated, may be a similar ipam like phpipam could be an alernative.
https://hub.docker.com/u/phpipam

When start to work with proxmox its easy to loose the control over the IP's and if you have tons of wifi iot devices its very difficult to manage the address space of your network. In this cases, a IPAM is the best solution because you can check the available ranges, IP's and take all your network under control

**Comment by derjoerg** [comment]
+1 for netbox

**Reply by c835722** [reply]
Further to this Netbox has a Proxmox integration plugin via https://github.com/netdevopsbr/netbox-proxbox which adds value to this script request.
Note that the Proxmox integration plugin required a VM instance rather than LXC to work currently.

**Reply by mdepedrof** [reply]
i dont know this pluging. sounds great! thanks for sharing!!

**Reply by MephistoaD** [reply]
This plugin, however, violates the best practices set by the netbox community: the netbox shall be configured manually, and all other data should be created by applying its version of the truth.

The "netbox version" of an installation is that you create a new vm entry in the netbox, asign it, for example, the role homeassistant, and then your configuration management (ansible, puppet, chef) will go out and create that vm in proxmox. 

This is an entirely different workflow to deploy things. It's not better, nor worse, but just different.

**Reply by c835722** [reply]
I take your point and thank you for sharing a key Netbox insight. 

For home-lab folks there is a large utility value I believe in having git controlled and supported community Netbox VM such that a greater number of the available plugins (which may indeed break best practice) be supported so that they can may be experimented with from a self education perspective. This netbox-promox plugin in one such plugin.

A ttdeck Netbox VM would further promote both Proxmox and Netbox usage and education I feel.

**Reply by mx4k** [reply]
@tteck Is there a chance that you will actually create a script for Netbox? Just asking because (tbh) I'm lazy. Really appreciate your work and before I try the install myself....

**Comment by McSego1990** [comment]
+1 phpipam

**Comment by Subnet-Zero** [comment]
+1 for netbox
